From a0fd9a9acbe3b4b5f6ba968ee4d053c0bb528240 Mon Sep 17 00:00:00 2001
From: Camber Huang <camber@poi.science>
Date: Thu, 28 Apr 2022 15:03:09 +0800
Subject: [PATCH] Forcibily enable compile support for mips
 (mips64le/loongson3)

Currently Avisynth+ doesn't support building for mips architectures, leading
to FTBFS for mips64le/loongson3. This patch will be a workaround to solve
aforementioned issue.

After doing some researches, it is discovered that some of supported
architectures doesn't contain any arch-specific code. Therefore I
concluded that it can be successfully built on loongson3 without any
arch-specific code by forcibily enable build support.

Signed-off-by: Camber Huang <camber@aosc.io>
---
 avs_core/include/avs/config.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/avs_core/include/avs/config.h b/avs_core/include/avs/config.h
index 464dc532..bdabf17f 100644
--- a/avs_core/include/avs/config.h
+++ b/avs_core/include/avs/config.h
@@ -61,6 +61,8 @@
 #   define RISCV
 #elif defined(__sparc_v9__)
 #   define SPARC
+#elif defined(__mips__)
+#   define MIPS
 #else
 #   error Unsupported CPU architecture.
 #endif
-- 
2.36.0

